### DISCUSSION TITLE
Global test context

### DISCUSSION THREAD

**Discussion by canassa** [vitest-dev/vitest#6222] [Q&A] [answered]
I am currently looking into the [test context documentation](https://vitest.dev/guide/test-context.html) and I have two questions:

1- Importing an extended helper like this is a bit cumbersome:

```typescript
import { expect } from 'vitest'
import { myTest } from './my-test.js'

myTest('add items to todos', ({ todos }) => {
  expect(todos.length).toBe(3)

  todos.push(4)
  expect(todos.length).toBe(4)
})
``` 

Is there a to make the extended context automatically available when importing directly from `vitest`?

2- The docs mention about extending `test`, but our tests are using a mixture of `test`, `describe`, `it`, etc. What's the best way to make the extended context available to all of these?

**Comment by sheremet-va** [comment, marked as answer]
1. It is not possible to extend the default `test` from `vitest`. I don't see how importing it is cumbersome, it takes the same amount of steps - I would recommend exporting a `test` function instead of `myTest`:
```ts
import { test as baseTest } from 'vitest'
export const test = baseTest.extend({
   // ...
})
```
2. `describe` doesn't care about the context so you can just use it. `it` is an alias for `test`, they reference the same function, so it doesn't matter which one you use
